x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -153,6 +157,55 @@ func (k Keeper) SetWithdrawalAddressOnHost(ctx sdk.Context, hostZone types.HostZ
 	return nil
 }
 
+func (k Keeper) ClaimAccruedStakingRewardsOnHost(ctx sdk.Context, hostZone types.HostZone) error {
+	// Fetch the relevant ICA
+	if hostZone.DelegationIcaAddress == "" {
+		return errorsmod.Wrapf(types.ErrICAAccountNotFound, "delegation ICA not found for %s", hostZone.ChainId)
+	}
+	if hostZone.WithdrawalIcaAddress == "" {
+		return errorsmod.Wrapf(types.ErrICAAccountNotFound, "withdrawal ICA not found for %s", hostZone.ChainId)
+	}
+	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "Withdrawal Address: %s, Delegator Address: %s",
+		hostZone.WithdrawalIcaAddress, hostZone.DelegationIcaAddress))
+
+	validators := hostZone.Validators
+
+	// Build multi-message transaction to withdraw rewards from each validator
+	// batching txs into groups of ClaimRewardsICABatchSize messages, to ensure they will fit in the host's blockSize
+
+	// Calculate the number of batches
+	numBatches := (len(validators) + ClaimRewardsICABatchSize - 1) / ClaimRewardsICABatchSize
+
+	// Iterate over the batches
+	for batch := 0; batch < numBatches; batch++ {
+		start := batch * ClaimRewardsICABatchSize
+		end := (batch + 1) * ClaimRewardsICABatchSize
+		if end > len(validators) {
+			end = len(validators)
+		}
+		msgs := []proto.Message{}
+		// Iterate over the items within the batch
+		for _, val := range validators[start:end] {
+			// skip withdrawing rewards
+			if val.Delegation.IsZero() {
+				continue
+			}
+			msg := &distributiontypes.MsgWithdrawDelegatorReward{
+				DelegatorAddress: hostZone.DelegationIcaAddress,
+				ValidatorAddress: val.Address,
+			}
+			msgs = append(msgs, msg)
+		}
+
+		_, err := k.SubmitTxsStrideEpoch(ctx, hostZone.ConnectionId, msgs, types.ICAAccountType_DELEGATION, "", nil)
```

Comment:
nit: because we skip assembling the msg if `val.Delegation.IsZero`, should we have a check here of `if len(msgs) > 0` then submit the Txs, and otherwise just continue with the loop?

Reply:
Good eye, added [here](https://github.com/Stride-Labs/stride/pull/961/commits/8b9393ac26fecb329ccc51f62eb5c69d05dd9969).